### PR TITLE
docbook-xsl: use brewed xmlcatalog on linux

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
I'm confused as to why this formula needs bottles. The install method looks like it's just moving some files around.

**Comment by SeekingMeaning** [comment]
Related: https://github.com/Homebrew/homebrew-core/pull/66910

**Comment by iMichka** [comment]
> I'm confused as to why this formula needs bottles. The install method looks like it's just moving some files around.

That's a different, unrelated topic. Can we focus on the PR? :)

> Related: #66910

So the idea is to use `ENV.prepend_path "PATH", Formula["libxml2"].opt_bin` instead?

File: Formula/docbook-xsl.rb [hunk 1]
```diff
@@ -61,6 +61,11 @@ def install
   end
 
   def post_install
+    xmlcatalog = "xmlcatalog"
+    on_linux do
+      xmlcatalog = Formula["libxml2"].opt_bin/"xmlcatalog"
```
Comment (carlocab) [inline Formula/docbook-xsl.rb:66]: Don't you need a
```ruby
uses_from_macos "libxml2"
```
for this to work? And if you add that, `xmlcatalog` should be in your `PATH` (right?) so this shouldn't be needed.

**Comment by carlocab** [comment]
> That's a different, unrelated topic. Can we focus on the PR? :)

The PR has a "bottle unneeded" label, so it looks like whether a bottle is needed or not is related to this PR. :)

**Comment by Bo98** [comment]
I tagged it like that because Linux-only changes do not require a new bottle on macOS. I'd really like to avoid the unnecessary purging of older bottles (maybe we should make `linux to homebrew-core` imply `automerge-skip`, and people can manually `brew pr-publish` if the change really does warrant new macOS bottles).

> I'm confused as to why this formula needs bottles. The install method looks like it's just moving some files around.

1 download vs 4 downloads I guess.

**Comment by iMichka** [comment]
Same as for `docbook, I removed the workaround from linuxbrew-core as the code was still present there: https://github.com/Homebrew/linuxbrew-core/pull/22648
It looks like it works without it, at least on CI.

Closing here, but feel free to discuss the bottle vs no bottle situation :)

**Pull request closed without merging**